lib/date_time/Kconfig

```diff
@@ -51,6 +51,19 @@ config DATE_TIME_NTP_QUERY_TIME_SECONDS
 	int "Duration in which the library will query for NTP time, in seconds"
 	default 5
 
+config DATE_TIME_RETRIES
```

Comment:
The name of the option could be `DATE_TIME_RETRY_COUNT`. This would make it more obvious that these two options are related. `_RETRY_COUNT`/`_RETRY_CNT` (I prefer the full name) is used in several places at least in Zephyr.